tests/console/systemd_testsuite.pm

```diff
@@ -0,0 +1,53 @@
+# SUSE's systemd tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run testsuite included in systemd sources
+# Maintainer: Thomas Blume <tblume@suse.com>
+
+use base "consoletest";
+use warnings;
+use strict;
+use testapi;
+use utils;
+
+sub run() {
+    # install systemd testsuite
+    select_console 'root-console';
+    if (check_var('VERSION', '12-SP2')) {
+        zypper_call('ar http://download.suse.de/ibs/QA:/SLE12SP2/update/ systemd-testrepo');
+    }
+
+    elsif (check_var('VERSION', '12-SP3')) {
+        zypper_call('ar http://download.suse.de/ibs/QA:/SLE12SP3/update/ systemd-testrepo');
+    }
+    else {
```

Comment:
IMHO our primary goal should be "user acceptance and system tests". Testsuites for certain packages can of course also be run within the context of openQA. I do not see a problem with test execution delaying the publishing of packages as the purpose of tests should be to ensure that there are no regressions, right?
One potential problem with openQA tests is that mainly QA engineers are reviewing the test results and there is a certain cost involved in a QA engineer reviewing a test failure, identifying the component that is potentially causing the failure and identifying the persons that could help with fixing. And then it takes a whole lot of a time to see the fix being active in the full products, i.e. new SLE builds, when the only thing you want to test is a patch to a single package. That is what I consider costly so I prefer, if possible, test suites in the package itself. If you like, take a look into my proof-of-concept of OBS multibuild package tests in https://build.opensuse.org/project/show/home:okurz:multibuild_package_test and I would appreciate feedback. I share coolo's opinion but only when it relates to system testing, e.g. booting a gnome session, checking integration of firefox with other applications, configuration of standard system services with yast or systemctl, etc.
But do not see my comment as a block to your PR just a mention for consideration.